### PR TITLE
New Google symptoms signals - omicron related

### PR DESCRIPTION
**Pull request by nloliveira**
### Description
Alterations in Google symptoms signals. 

Removed: anosmia, ageusia, sum_anosmia_ageusia
Added: S01, S02, S03, S05, S06, S08, SControl

New signals are averages of symptom trends, while old sum_anosmia_ageusia signal was a sum. None of our signals are related to only one symptom, as was the case with anosmia and ageusia signals. 

Replaced old smoothers (form smooth.py) for new smoothers (Smooth class from delphi_utils). 

Adapted unit tests for multiple combinations of symptoms and new smoothers.

### CODE REVIEW

**Comment by krivard** [comment]
Hmm that test shouldn't be failing, but it looks like this branch is behind main. Would you try this:
```
$ git checkout google_symptoms_omicron
$ git merge main
$ git push origin google_symptoms_omicron
```

File: google_symptoms/delphi_google_symptoms/run.py [hunk 1]
```diff
@@ -94,15 +94,18 @@ def run_module(params):
...
-            df = df_pull.set_index(["timestamp", "geo_id"])
-            df["val"] = df[metric].groupby(level=1
-                                           ).transform(SMOOTHERS_MAP[smoother][0])
+            df = df_pull#.set_index(["timestamp", "geo_id"])
+            df["val"] = df[metric].astype(float)
+            df["val"] = df[["geo_id", "val"]].groupby(
+                "geo_id")["val"].transform(
+                SMOOTHERS_MAP[smoother][0].smooth)
+            #df["val"] = df[metric].groupby(level=1
+            #                               ).transform(SMOOTHERS_MAP[smoother][0].smooth)
```
Comment (nmdefries) [inline google_symptoms/delphi_google_symptoms/run.py:108]: Please remove out-of-date code.

```suggestion
            df = df_pull
            df["val"] = df[metric].astype(float)
            df["val"] = df[["geo_id", "val"]].groupby(
                "geo_id")["val"].transform(
                SMOOTHERS_MAP[smoother][0].smooth)
```

**Review by nmdefries — CHANGES_REQUESTED** [review]
Some questions and a few code cleanup things.

We'll also need to add all smoothed signal names to validation params in local and ansible `params.json`.

File: google_symptoms/tests/test_data/bad_county_invalid_fips.csv [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
...pse,Red eye,Renal colic,Restless legs syndrome,Rheum,Rhinitis,Rhinorrhea,Rosacea,Round ligament pain,Rumination,Scar,Sciatica,Scoliosis,Seborrheic dermatitis,Self harm,Sensitivity to sound,Sexual dysfunction,Shallow breathing,Sharp pain,Shivering,Shortness of breath,Shyness,Sinusitis,Skin condition,Skin rash,Skin tag,Skin ulcer,Sleep apnea,Sleep deprivation,Sleep disorder,Snoring,Sore throat,Spasticity,Splenomegaly,Sputum,Stomach rumble,Strabismus,Stretch marks,Stridor,Stroke,Stuttering,Subdural hematoma,Suicidal ideation,Swelling,Swollen feet,Swollen lymph nodes,Syncope,Tachycardia,Tachypnea,Telangiectasia,Tenderness,Testicular pain,Throat irritation,Thrombocytopenia,Thyroid nodule,Tic,Tinnitus,Tonsillitis,Toothache,Tremor,Trichoptilosis,Tumor,Type 2 diabetes,Unconsciousness,Underweight,Upper respiratory tract infection,Urethritis,Urinary incontinence,Urinary tract infection,Urinary urgency,Uterine contraction,Vaginal bleeding,Vaginal discharge,Vaginitis,Varicose veins,Vasculitis,Ventricular fibrillation,Ventricular tachycardia,Vertigo,Viral pneumonia,Visual acuity,Vomiting,Wart,Water retention,Weakness,Weight gain,Wheeze,Xeroderma,Xerostomia,Yawn,hyperhidrosis,pancreatitis,geo_id
```
[the remark this answers is not in the document]
Comment (nloliveira) [reply google_symptoms/tests/test_data/bad_county_invalid_fips.csv:1]: Yes, we noticed this problem with the failing assertion. While it isn't a problem now, I agree we should keep an eye on this.

File: google_symptoms/delphi_google_symptoms/constants.py [hunk 3]
```diff
@@ -1,15 +1,88 @@
...
+    "s03": ["Fever", "Hyperthermia", "Chills", "Shivering", "Low grade fever"],
+    #"s04": ["Fatigue", "Weakness", "Muscle weakness", "Myalgia", "Pain"],
+    "s05": ["Shortness of breath", "Wheeze", "Croup", "Pneumonia", "Asthma",
+            "Crackles", "Acute bronchitis", "Bronchitis"],
+    "s06": ["Anosmia", "Dysgeusia", "Ageusia"],
+    #"s07": ["Nausea", "Vomiting", "Diarrhea", "Indigestion", "Abdominal pain"],
+    "s08": ["Laryngitis", "Sore throat", "Throat irritation"],
+    #"s09": ["Headache", "Migraine", "Cluster headache", "Dizziness", "Lightheadedness"],
+    #"s10": ["Night sweats","Perspiration", "hyperhidrosis"],
+    "scontrol": ["Type 2 diabetes", "Urinary tract infection", "Hair loss",
```
[the remark this answers is not in the document]
Comment (nloliveira) [reply google_symptoms/delphi_google_symptoms/constants.py:21]: Makes sense. Nigam put the control set together, we can double check with him.

**Comment by nloliveira** [comment]
@nmdefries thank you, code is updated! I also added the new names to the json template file, so it just needs the API key. 

There is one failing check that it doesn't seem to be related to our code; it's a unit test from the hhs pipeline. Do you know what that could be?

File: google_symptoms/params.json.template [hunk 4]
```diff
@@ -26,9 +26,13 @@
...
-        "ageusia_smoothed_search",
-        "sum_anosmia_ageusia_smoothed_search",
-        "anosmia_smoothed_search"
+        "s01_smoothed_search",
+        "s02_smoothed_search",
+        "s03_smoothed_search",
+        "s05_smoothed_search",
+        "s06_smoothed_search",
+        "s08_smoothed_search",
+        "scontrol_smoothed_search"
```
Comment (nmdefries) [inline google_symptoms/params.json.template:35]: Please apply the same changes to the production params file, [here](https://github.com/cmu-delphi/covidcast-indicators/blob/20da1c0c6da1f2886e87b88b671f8247e8a36fad/ansible/templates/google_symptoms-params-prod.json.j2).

The API key is already accessible to the production params, so we don't need to do anything there.

Comment (nloliveira) [reply google_symptoms/params.json.template:35]: Thanks for the pointer! I didn't know I had access to the production params file.
I just pushed the updated file.

**Review by nmdefries — APPROVED** [review]
Looks good!

I'm waiting on some people to look at the HHS test failure. Will let you know when that's resolved and then we can merge `main` back into this branch to update the tests here.

**Comment by nmdefries** [comment]
Okay, the hhs tests are fixed on `main`, you can merge that in now.

**Comment by nmdefries** [comment]
@korlaxxalrok This is ready to merge and release.

**Pull request merged**